examples/todoapp/test/feature_todo_management/blocs/todo_management_test.dart

```diff
@@ -0,0 +1,110 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:rx_bloc_test/rx_bloc_test.dart';
+import 'package:todoapp/base/common_blocs/coordinator_bloc.dart';
+import 'package:todoapp/base/common_services/todo_list_service.dart';
+import 'package:todoapp/base/models/errors/error_model.dart';
+import 'package:todoapp/base/models/todo_model.dart';
+import 'package:todoapp/feature_todo_management/blocs/todo_management_bloc.dart';
+import 'package:todoapp/feature_todo_management/services/todo_manage_service.dart';
+import 'package:todoapp/feature_todo_management/services/todo_validator_service.dart';
+import 'package:todoapp/lib_router/blocs/router_bloc.dart';
+
+import '../../base/common_blocs/coordinator_bloc_mock.dart';
+import '../../base/common_blocs/router_bloc_mock.dart';
+import '../../stubs.dart';
+import 'todo_management_test.mocks.dart';
+
+@GenerateMocks([
+  TodoListService,
+  TodoManageService,
+  TodoValidatorService,
+  RouterBlocType,
+  TodoManagementBlocEvents,
+  TodoManagementBlocStates,
+  TodoManagementBlocType,
+])
+void main() {
+  late TodoListService _listService;
+  late TodoManageService _todoManageService;
+  late TodoValidatorService _validatorService;
+  late CoordinatorBlocType _coordinatorBloc;
+
+  void _defineWhen(
+      {String? todoId,
+      String? title,
+      String? description,
+      TodoModel? todoModel}) {
+    when(_listService.fetchTodoById(todoId ?? '')).thenAnswer((_) {
+      if (todoId?.isNotEmpty != null) {
+        return Future.value(todoModel);
+      }
+
+      return Future.error(Stubs.notFoundError);
+    });
+
+    when(_todoManageService.addOrUpdate(Stubs.todoEmpty
+            .copyWith(title: title, description: description ?? '')))
```

Comment:
Be aware that the defaults values can be added to the method definition, which will improve the visibility